### DISCUSSION TITLE
Enable / disable motion and alerts from mqtt

### DISCUSSION THREAD

**Discussion by ilium007** [tsightler/ring-mqtt#516] [Support Q&A] [answered]
Just wanting to know if there is any way to enable / disable alerts from an mqtt topic? I would like to use the presence detection I have in HomeAssistant /Nodered to disable and enable alerts and motion recording.

**Comment by gbdavidx** [comment]
Motion sensor is not the same as presence detection

**Reply by ilium007** [reply]
> Motion sensor is not the same as presence detection
> […](#)

Yeah mate - I understand that. My homeassistant presence detection is more accurate for me than the Ring geolocation which sometimes lags. I want to disable Ring alerts and Ring motion detection based on the state of my Homeassistant presence.

**Comment by tsightler** [comment, marked as answer]
Right now it's not possible although it's "on my list", although, as is probably obvious, I haven't had any time to really work on features for this project lately.  What little time I've had is mostly focused on fixing bugs and trying to keep things that already exist working.

I think you could probably implement this with modes.  Setup your cameras home and away modes to enable notifications/recording as you wish and then have HA automatically set home or away mode based on an automation.  Setting modes is possible today but you have to enable the modes feature explicitly in the config.

**Reply by ilium007** [reply]
That would work for me I think! I’ll take a look.

**Reply by ilium007** [reply]
I enabled the config option (enable_modes) and I can see the mqtt topic changing when I change the setting in the phone app but when I set the topic (ring/xxx/alarm/xxx_mode/mode/state) to "disabled" the mode doesn't change in the iOS app and within seconds the mode reverts in the mqtt topic.

**Reply by ilium007** [reply]
Disregard! I didn't quite understand what an 'alarm panel' was in HA. All working!

<img width="510" alt="image" src="https://user-images.githubusercontent.com/732766/206313121-eb85b237-3dcb-40c1-9706-bebb08f91f8a.png">

**Reply by ilium007** [reply]
Also worked out that the way to change using mqtt messages is using the topic `ring/xxx/alarm/xxx_mode/mode/command` and the modes: `DISARM` `ARM_HOME` `ARM_AWAY`

**Reply by tsightler** [reply]
Technically you shouldn't have to work it out as the topics and values are [documented](https://github.com/tsightler/ring-mqtt/wiki/MQTT-Device-Topics#modes-control-panel).  If the documentation is wrong or incorrect, feel free to let me know.
